### PR TITLE
Add numProblems as an updatable room setting

### PR DESCRIPTION
**Pull request by alankbi**
### List of Changes:

* Add numProblems as a customizable field to room objects
* Use numProblems setting from rooms to initialize game objects
* Create potential fixes for the invalid numProblems errors 
  * Set a max of 10 numProblems
  * If numProblems is more than total problems than exist, return as many problems as exists
  * Prevent room from being active if game creation fails

### Notes:

* I didn't add the ability to customize numProblems on the frontend since for the MVP I'm guessing we'll only be able to support solving one problem per game (the first problem) - let me know what your thoughts are on that

### CODE REVIEW

**Comment by alankbi** [comment]
[Allow numProblems to be customized](https://trello.com/c/vs9UC6ta/216-allow-numproblems-to-be-customized)

File: src/main/java/com/rocketden/main/service/ProblemService.java [hunk 1]
```diff
@@ -92,10 +92,15 @@ public ProblemDto getProblem(String problemId) {
             throw new ApiException(ProblemError.NOT_FOUND);
```
Comment (alankbi) [inline src/main/java/com/rocketden/main/service/ProblemService.java:92]: Not allowed to comment on it but thoughts on changing lines 78-82 to return a generic `ProblemError.EMPTY_FIELD` error if either setting is missing?

**Comment by zpChris** [comment]
> * I didn't add the ability to customize numProblems on the frontend since for the MVP I'm guessing we'll only be able to support solving one problem per game (the first problem) - let me know what your thoughts are on that

Yeah I agree, I think for the MVP it'll just be one problem at a time 👍

File: src/main/java/com/rocketden/main/service/ProblemService.java (hunk 1 above)
Comment (zpChris) [reply src/main/java/com/rocketden/main/service/ProblemService.java:92]: Yeah I'm ok with that 👍

**Review by zpChris — APPROVED** [review]
LGTM! No nits or comments at all 🎉 😃

**Comment by zpChris** [comment]
I merged #86 so I'd advocate pulling into this branch to ensure all the tests and whatnot still work 👍

**Pull request merged**